scripts/diff-pkg-changes.py

```diff
@@ -0,0 +1,351 @@
+                pkg_dict['Source'],
+                pkg_dict['Description'],
+                pkg_dict['OE'],
+                pkg_dict.get('HomePage', ''),
+                pkg_dict['License'],
+                pkg_dict['Priority']
+            ))
+        except KeyError:
+            raise UnexpectedError(
+                'Missing key info in Packages file of feed \'%s\' in branch %s.' \
+                    % (feed_name, branch_name)
+            )
+
+    return pkgs
+
+# Create a final report from all the package versions
+def gen_report(
+    old_core_pkgs: list['Package'], new_core_pkgs: list['Package'],
+    old_extra_pkgs: list['Package'], new_extra_pkgs: list['Package'],
+    old_vers: str, new_vers: str
+) -> dict:
+    # Handle core diffs
+    core_pkg_changes: list[dict] = []
+    core_pkgs_added: list[dict[str, str]] = []
+    core_pkgs_removed: list[dict[str, str]] = []
+    for pkg in old_core_pkgs:
+        other_vers_of_pkg: list['Package'] = \
+            [ i for i, other_pkg in enumerate(new_core_pkgs) if other_pkg.name == pkg.name ]
+        if len(other_vers_of_pkg) == 0: # Check for package removal from version to version
+            core_pkgs_removed.append({
+                'name': pkg.name,
+                'version': pkg.version
+            })
+        else:
+            other_pkg: 'Package' = new_core_pkgs[other_vers_of_pkg[0]] # We know there's only one FYI
+            diff: dict = pkg.diff(other_pkg)
+            if diff != {}:
+                core_pkg_changes.append(diff)
+            new_core_pkgs.pop(other_vers_of_pkg[0])
+    for pkg in new_core_pkgs:
+        # We removed packages that were duplicates, so what's left is new packages
+        core_pkgs_added.append({
+            'name': pkg.name,
+            'version': pkg.version
+        })
+
+    # Do the same for extra
+    extra_pkg_changes: list[dict] = []
+    extra_pkgs_added: list[dict[str, str]] = []
+    extra_pkgs_removed: list[dict[str, str]] = []
+    for pkg in old_extra_pkgs:
+        other_vers_of_pkg: list['Package'] = \
+            [ i for i, other_pkg in enumerate(new_extra_pkgs) if other_pkg.name == pkg.name ]
+        if len(other_vers_of_pkg) == 0: # Check for package removal from version to version
+            extra_pkgs_removed.append({
+                'name': pkg.name,
+                'version': pkg.version
+            })
+        else:
+            other_pkg: 'Package' = new_extra_pkgs[other_vers_of_pkg[0]] # We know there's only one FYI
+            diff: dict = pkg.diff(other_pkg)
+            if diff != {}:
+                extra_pkg_changes.append(diff)
+            new_extra_pkgs.pop(other_vers_of_pkg[0])
+    for pkg in new_extra_pkgs:
+        # We removed packages that were duplicates, so what's left is new packages
+        extra_pkgs_added.append({
+            'name': pkg.name,
+            'version': pkg.version
+        })
+
+    return {
+        (old_vers + '->' + new_vers): {
```

Comment:
I think its likely that we will want to change the contents of this data structure at some point in the future. Can we add a "version" field, which expresses the version of the data structure?